Site Selector: Remove `siteSlug` from 'Add new site' button link to prevent unexpected behavior

danielbachhuber opened the pull request:
## Proposed Changes

Switches from `siteSlug` to `sourceSiteSlug` to prevent a state collision where the Plans picker wouldn't load. Introduced in https://github.com/Automattic/wp-calypso/pull/70571

See p1670240472927109-slack-C029GN3KD

<img width="419" alt="Screenshot 2022-12-05 at 14 26 54" src="https://user-images.githubusercontent.com/87168/205637600-7e788f73-29aa-45d1-8969-5edde2ce6f1d.png">


## Testing Instructions

1. Make sure I didn't miss any other variables that need to be renamed.
2. From My Home Sidebar, click 'Switch site' and then 'Add new site'.
3. Verify the back button appears as 'Back to Sites'.
4. Enter a domain and pick the free `.wordpress.com` option.
5. Verify the plan picker loads.
6. Navigate to `/sites` in a new window and click 'Add new site'.
7. Verify the back button appears as 'Back to Sites'.
8. Enter a domain and pick the free `.wordpress.com` option.
4. Verify the plan picker loads.

Note: some lovely caching wonkiness can occur without refreshing the page: https://github.com/Automattic/wp-calypso/pull/68727#issuecomment-1271241893 We can ignore that.

danielbachhuber committed:
Site Selector: Switch to `sourceSiteSlug` to prevent collisions

simison reviewed (changes requested):
The change will have wider consequences that need to be mapped out and changed, too.

For example;
- Have an un-launched site
- Go to Settings, scroll to "launch site" section
    <img width="735" alt="image" src="https://user-images.githubusercontent.com/87168/205633734-5bead4e4-3742-47e6-81b8-7dc91b898654.png">

- Note the button has URL `https://wordpress.com/start/launch-site?siteSlug=SITE_SLUG&source=general-settings&hide_initial_query=yes`
- Domain step has "back to settings"
    <img width="471" alt="image" src="https://user-images.githubusercontent.com/87168/205633692-68b1b0fd-3abb-48e0-b931-18e2ca62c831.png">

In this branch that back link turns into "back to sites":

<img width="372" alt="image" src="https://user-images.githubusercontent.com/87168/205634386-97eb094b-31f4-4ddd-9e2c-a8da0c885a34.png">

`siteSlug` is probably important for the above launch flow just to know which site to launch, so editing is a bit more complex.

Wondering if there's something simpler that could fix this in interim, e.g. just remove the `siteSlug` from "add site" button? The back button isn't _that_ important; browsers have one natively after all. If it ends up showing "back to all sites" by removing the slug — well that's kinda correct. :-)

danielbachhuber committed:
Revert "Site Selector: Switch to `sourceSiteSlug` to prevent collisions"

This reverts commit 45ba6065a6f0fa12e5e7a1e6f23408000a6d9ff6.

danielbachhuber committed:
Simply remove the `siteSlug` to avoid causing other problems

danielbachhuber commented:
> The change will have wider consequences that need to be mapped out and changed, too.

@simison Oh, good point. I've reverted my original commit, and simply removed the `siteSlug` from the 'Add new site' button 896b7fae427042834926a96900ebc77396cd36dc

simison commented:
Thanks, that seems like good intermediate remedy for now. 👍  FYI I also added a screenshot to PR description for clarity for others.

Why the plans step got skipped just because `siteSlug` is there (and especially if that wasn't happening before) is still a concern. I looked at older PRs briefly and there were some kinda related changes in https://github.com/Automattic/wp-calypso/pull/70371 —  @southp @jdc91 who also are more familiar with Start framework than I am — anything comes to your mind what might've broken the "add site" button in first place?

simison reviewed (approved):
Tests well now:

- Adding site from the sidebar, `siteSlug` is gone
- I see plans grid after picking free domain
